dandi/misctypes.py

```diff
@@ -345,7 +345,10 @@ def open(self) -> IO[bytes]:
         # Optional dependency:
         import fsspec
 
-        return cast(IO[bytes], fsspec.open(self.url, mode="rb"))
+        # We need to call open() on the return value of fsspec.open() because
+        # otherwise the filehandle will only be opened when used to enter a
+        # context manager.
+        return cast(IO[bytes], fsspec.open(self.url, mode="rb").open())
```

Comment:
could there be a simple unittest for this functionality? it seems to be not covered by tests according to Codecov

Reply:
Done: cef7192.